maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/NewGradleJSFModuleProjectProvider.java

```diff
@@ -0,0 +1,224 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.maven.core;
+
+import com.liferay.ide.core.ILiferayProjectImporter;
+import com.liferay.ide.core.LiferayCore;
+import com.liferay.ide.maven.core.aether.AetherUtil;
+import com.liferay.ide.project.core.ProjectCore;
+import com.liferay.ide.project.core.modules.jsf.NewLiferayJSFModuleProjectOp;
+import com.liferay.ide.project.core.util.LiferayWorkspaceUtil;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.maven.archetype.ArchetypeGenerationRequest;
+import org.apache.maven.archetype.ArchetypeGenerationResult;
+import org.apache.maven.archetype.catalog.Archetype;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.m2e.core.internal.MavenPluginActivator;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.eclipse.m2e.core.project.ResolverConfiguration;
+import org.eclipse.sapphire.platform.PathBridge;
+
+/**
+ * @author Simon Jiang
+ */
+
+@SuppressWarnings( "restriction" )
+public class NewGradleJSFModuleProjectProvider extends NewMavenJSFModuleProjectProvider
+{
+
+    @Override
+    public IStatus createNewProject( NewLiferayJSFModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
+    {
+        IStatus retval = Status.OK_STATUS;;
+
+        try
+        {
+            final String projectName = op.getProjectName().content();
```

Comment:
you mean we use same LiferayModuleProjectOp? Because the provider is binded with different Op, I want to make the new Op extends from NewLiferayModuleProjectOp and try to reuse the existed code. Its better to make the their projectProvider seperarate.